tests/runner.pm

```diff
@@ -1066,7 +1066,10 @@ sub singletest_clean {
         }
     }
 
-    waitlockunlock($serverlogslocktimeout);
+    my @killtestservers = getpart("client", "killserver");
+    if(!@killtestservers) { # we are not killing it anyway
```

Comment:
What about the case where a test uses multiple servers but is only killing one?

Reply:
There is indeed a single test case that does that 1561 (and there might be future ones).

I'll submit a change in ordering here:
* execute the `postcommanddelay` first
* kill all specified servers, removing their lock files when necessary
* wait for any remaining servers to release their locks
* record the test timing